x-pack/plugins/lens/server/routes/field_stats.ts

```diff
@@ -205,6 +209,81 @@ export async function getNumberHistogram(
   };
 }
 
+export async function getNumberOnlyHistogram(
```

Comment:
nit: It looks like this code is almost the same as `getNumberHistogram` (except for the top values part). Can we reduce the redundancy somehow in a nice way (maybe some conditional branches)? Not a blocker, just stumbled over it a little.